ext/socket/lib/socket.rb

```diff
@@ -623,45 +692,130 @@ def accept_nonblock(exception: true)
   def self.tcp(host, port, local_host = nil, local_port = nil, connect_timeout: nil, resolv_timeout: nil) # :yield: socket
     last_error = nil
     ret = nil
+    attempt_v6 = true
+    attempt_v4 = true
+    start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+
+    if connect_timeout
+      raise ArgumentError, "connect_timeout must be Numeric" unless connect_timeout.is_a?(Numeric)
+      raise ArgumentError, "connect_timeout must not be negative" if connect_timeout.negative?
+    end
 
     local_addr_list = nil
-    if local_host != nil || local_port != nil
+    if local_host && local_port
```

Comment:
This ignore `local_host` and `local_port` if only one of them is specified.
Currently, `Socket.tcp`  can be used with only with `local_host` or `local_port` such as:
`Socket.tcp("127.0.0.1", 9000, "192.168.42.6")` or
`Socket.tcp("127.0.0.1", 9000, nil, 9001)`.

Reply:
I fixed this.